### PR TITLE
Manually instantiate cloned cells rather than deepcopying

### PR DESCRIPTION
**Pull request by joshmay1**
These proposed changes replace the deepcopying of cloned cells with a manual new instantiation. Any present optional cell properties are applied to the clone. Depending on cloning settings, the fill and region properties are set accordingly as before.

This provides a noticeable and significant performance improvement, particularly when cloning complicated cells, because it removes the potential for extra steps. With the deepcopy method, the original cell's fill and region attributes were deepcopied and immediately overwritten.

I've added some more testing to check this implementation, please let me know if anything looks like it might be off!

### CODE REVIEW

File: openmc/cell.py [hunk 1]
```diff
@@ -519,8 +518,15 @@ def clone(self, clone_materials=True, clone_regions=True,  memo=None):
             paths = self._paths
             self._paths = None
 
-            clone = deepcopy(self)
-            clone.id = None
+            clone = openmc.Cell()
+            clone.name = self.name
```
Comment (paulromano) [inline openmc/cell.py:522]:
```suggestion
            clone = openmc.Cell(name=self.name)
```

File: tests/unit_tests/test_cell.py [hunk 2]
```diff
@@ -57,25 +57,51 @@ def test_clone():
...
     c4 = c.clone(clone_regions=False)
     assert c4.id != c.id
     assert c4.fill != c.fill
     assert c4.region == c.region
-    assert c4.temperature == c.temperature
+
+    # Add optional properties to the original cell to ensure they're cloned successfully
+    c.temperature = 650.
+    c.translation = [1,2,3]
+    c.rotation = [4,5,6]
```
Comment (paulromano) [inline tests/unit_tests/test_cell.py:80]:
```suggestion
    c.translation = (1., 2., 3.)
    c.rotation = (4., 5., 6.)
```

File: tests/unit_tests/test_cell.py [hunk 3]
```diff
@@ -57,25 +57,51 @@ def test_clone():
...
+    assert c5.temperature == c.temperature
+    assert c5.volume == c.volume
+    assert all(c5.translation == c.translation)
+    assert all(c5.rotation == c.rotation)
+
+    # Mutate the original to ensure the changes are not seen in the clones
+    c.fill = openmc.Material()
+    c.region = +openmc.ZCylinder()
+    c.translation = [-1,-2,-3]
+    c.rotation = [-4,-5,-6]
```
Comment (paulromano) [inline tests/unit_tests/test_cell.py:96]:
```suggestion
    c.translation = (-1., -2., -3.)
    c.rotation = (-4., -5., -6.)
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for another performance improvement @joshmay1! Only a few trivial comments:

**Comment by joshmay1** [comment]
You're welcome @paulromano and thanks for the review! Similar changes can be made to universe cloning as well. I'll get a PR up with those soon.

**Review by paulromano — APPROVED** [review]

**Pull request merged**